library/src/main/java/pl/netigen/premium/PremiumFragmentNetigenApi.java

```diff
@@ -0,0 +1,147 @@
+package pl.netigen.premium;
+
+
+import android.graphics.Point;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.bumptech.glide.Glide;
+
+
+import java.util.List;
+
+import androidx.appcompat.widget.AppCompatTextView;
+import androidx.fragment.app.Fragment;
+import androidx.viewpager.widget.ViewPager;
+import pl.netigen.netigenapi.R;
+
+
+public class PremiumFragmentNetigenApi extends Fragment {
+
+
+    private int bgImage;
+    private int titleImage;
+    private int bgPremiumItem;
+    private int imageButtonBuy;
+    private int premiumOptionsTitle;
+    private int textBuyButton;
+
+    private List<PremiumItemNetigenApi> premiumItemNetigenApis;
+    private CarouselEffectTransformer transformer;
+    private View.OnClickListener clickBuyPremium;
+
+
+    private PremiumBuilderNetigenApi premiumBuilder;
+    private ViewPager viewPager;
+    private ImageView backgroundPremium;
+    private ImageView premiumTitleImage;
+    private TextView buyPremiumButton;
+    private AppCompatTextView textAboutPremium;
+
+    private int pageMargin;
+    private int viewPagerPadding;
+
+    public static PremiumFragmentNetigenApi newInstance(PremiumBuilderNetigenApi premiumBuilder) {
+        PremiumFragmentNetigenApi fragment = new PremiumFragmentNetigenApi();
+        fragment.setArguments(premiumBuilder);
+        return fragment;
+    }
+
+    private void setArguments(PremiumBuilderNetigenApi premiumBuilder) {
+        this.premiumBuilder = premiumBuilder;
+    }
+
+
+    public PremiumFragmentNetigenApi() {
+    }
+
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View inflate = inflater.inflate(R.layout.fragment_premium_netigen_api, container, false);
+        initValues();
+        bindView(inflate);
+        addImageAndText();
+        calculationsValues();
+        createViewPager();
+        return inflate;
+    }
+
+    private void initValues() {
+        bgImage = premiumBuilder.bgImage;
+        titleImage = premiumBuilder.titleImage;
+        bgPremiumItem = premiumBuilder.bgPremiumItem;
+        imageButtonBuy = premiumBuilder.imageButtonBuy;
+        premiumOptionsTitle = premiumBuilder.premiumOptionsTitle;
+        textBuyButton = premiumBuilder.textBuyButton;
+        premiumItemNetigenApis = premiumBuilder.premiumItemNetigenApis;
+        clickBuyPremium = premiumBuilder.clickBuyPremium;
+    }
+
+    private void calculationsValues() {
+        float density = getResources().getDisplayMetrics().density;
+        int widthScale = getResources().getInteger(R.integer.widthScale);
```

Comment:
Names widthScale or marginScale are too generic might cause conflicts with other apps resources. Should be more describful.